internal/cmp/normalize.go

```diff
@@ -94,6 +94,11 @@ func traverseValue(value reflect.Value, f func(slice reflect.Value)) {
 		}
 
 	case reflect.Slice:
+		// initialize empty slices to nil for normalization to be comparable
+		if value.Len() == 0 && value.CanSet() {
+			value.Set(reflect.Zero(value.Type()))
+		}
```

Comment:
Why not the other way around, I mean, nil is not iterable, while empty it. Empty arrays are usually less problematic.

nit pick question, BTW it is fine by me either way.

Reply:
> nil is not iterable, while empty it

`nil` indeed is iterable

Reply:
https://go.dev/play/p/murKIez6rpW

Reply:
TIL 
Thanks!